eventprocessor/handlers.go

```diff
@@ -1,61 +1,145 @@
 package eventprocessor
 
 import (
-    "context"
-    "encoding/json"
-    eventlistener "github.com/DaoCasino/platform-action-monitor-client"
-    "github.com/eoscanada/eos-go"
-
-    "github.com/rs/zerolog/log"
+	"context"
+	"encoding/json"
+	eventlistener "github.com/DaoCasino/platform-action-monitor-client"
+	"github.com/eoscanada/eos-go"
+	"github.com/rs/zerolog/log"
+	"platform-backend/models"
+    "time"
 )
 
-func onGameStarted(ctx context.Context, p *EventProcessor, event *eventlistener.Event) error {
-    return nil
-}
+func onGameStarted(ctx context.Context, p *EventProcessor, event *eventlistener.Event, session *models.GameSession) error {
+	log.Debug().Msgf("Got started event for session: %d", session.ID)
 
-func onActionRequest(ctx context.Context, p *EventProcessor, event *eventlistener.Event) error {
-    return nil
-}
+    err := p.repos.GameSession.AddGameSessionUpdate(ctx, &models.GameSessionUpdate{
+        SessionID:  session.ID,
+        UpdateType: models.SessionStartedUpdate,
+        Timestamp:  time.Now(),
+        Data:       event.Data,
+    })
+    if err !=  nil {
+    	return err
+	}
 
-func onSignidicePartOneRequest(ctx context.Context, p *EventProcessor, event *eventlistener.Event) error {
-    gs, err := p.repos.GameSession.GetGameSession(ctx, event.RequestID)
-    if err != nil {
-        return err
-    }
+	err = p.repos.GameSession.UpdateSessionState(ctx, session.ID, models.GameStartedInBC)
+	if err !=  nil {
+		return err
+	}
 
-    game, err := p.repos.Casino.GetGame(ctx, gs.GameID)
-    if err != nil {
-        return err
-    }
+	return nil
+}
 
-    var data struct{ Digest eos.Checksum256 }
-    err = json.Unmarshal(event.Data, &data)
-    if err != nil {
-        return err
-    }
+func onActionRequest(ctx context.Context, p *EventProcessor, event *eventlistener.Event, session *models.GameSession) error {
+	log.Debug().Msgf("Got action request event for session: %d", session.ID)
 
-    err = p.useCases.Signidice.PerformSignidice(ctx, game.Contract, data.Digest, gs.BlockchainSesID)
-    if err != nil {
-        return err
+	// first action already processed by back-end
+    if session.State == models.GameStartedInBC {
+        return nil
     }
 
-    log.Debug().Msgf("Successfully signed and sent signidice for session: %d", gs.ID)
+    err := p.repos.GameSession.AddGameSessionUpdate(ctx, &models.GameSessionUpdate{
```

Comment:
i think we should rename the method to `AddSessionUpdate` - `GameSession` already implies it is a a game session update
Similar with EventProcessor

Reply:
rename AddSessionUpdate to Update ?

Reply:
just bringing attention, no need to rename in this PR